quinn-proto/src/connection/streams/mod.rs

```diff
@@ -129,11 +129,13 @@ impl<'a> RecvStream<'a> {
     /// Discards unread data and notifies the peer to stop transmitting. Once stopped, further
     /// attempts to operate on a stream will yield `UnknownStream` errors.
     pub fn stop(&mut self, error_code: VarInt) -> Result<(), UnknownStream> {
-        let mut entry = match self.state.recv.entry(self.id) {
-            hash_map::Entry::Occupied(s) => s,
-            hash_map::Entry::Vacant(_) => return Err(UnknownStream { _private: () }),
-        };
-        let stream = entry.get_mut();
+        let stream =
+            match self.state.recv.get_mut(&self.id).map(|s| {
+                s.get_or_insert_with(|| Box::new(Recv::new(self.state.stream_receive_window)))
+            }) {
+                Some(s) => s,
+                None => return Err(UnknownStream { _private: () }),
+            };
```

Comment:
It's not clear to me why the old `Entry` logic had to go away here. Couldn't we slap a `let stream = stream,get_or_insert_with(...)` after the existing code to the same effect?